### ISSUE TITLE
[SignalR Service Support] Client to support invocation messages from send channel

### ISSUE THREAD

**Issue by vicancy** [SignalR/SignalR#4201]
[Group functional tests](https://github.com/SignalR/SignalR/blob/anurse/service-functional-tests/test/Microsoft.AspNet.SignalR.Client.JS.Tests/wwwroot/Tests/FunctionalTests/Hubs/HubGroupFacts.js) with SSE and LongPolling fail  , the root cause is: the `group.join` never receives a callback `done()`. This is because for SSE and LongPolling, `connection.onReceived` is never triggered for Invocation Messages such as `{"I": "0"}` 

For SSE and LongPolling, they have 2 channels, one for send and one for receive, persistent responses come back on the receive channel, but innovation responses come back on the send channel. Server change is pretty hard to make, hopefully we can support it from the client side.

cc @anurse @davidfowl

**Comment by anurse** [comment]
At first glance, I think this can actually be resolved by changing the test hub here. The hub method is void-returning and calls Groups.Add which is an async operation that returns a `Task`. We just need the code to wait for the add to complete before returning (I don't know off-hand if SignalR hubs support returning Tasks, I'll look in to that).

**Comment by vicancy** [comment]
@anurse Updated the issue with more info.

**Comment by anurse** [comment]
Yeah, I caught up with @davidfowl this morning and he filled me in. I'll look at this
